content/root/best-practice-guide.textile

```diff
@@ -19,28 +19,57 @@ h2(#ably-101). Ably 101
+
+**Important:** Do not use token authetication on the server, as this will result in unnecessary overhead - the server will periodically require token requests from Ably to authenticate. Simply use basic authentication server-side.
+
+If you prefer to use a single auth strategy across all third-party services, you can use "JWT Authentication":/core-features/authentication#ably-jwt-process as part of Token authentication.
+
+h3(#using-api-key-client-side). Using the API key on the client side
+
+Ably's "Basic authentication":/core-features/authentication#basic-authentication method is the easiest way to authenticate your clients on the Ably platform, but, this will directly expose the API key in the source code. Hence, when using it on the client side, any entity that interacts with the client could view your API key and potentially misuse it. So, if you can run an auth server, we recommend using "Token authentication":/core-features/authentication#token-authentication instead. With this method, your auth server is expected to use the API key to raise a token request from Ably, and the token can then be used by your end clients in the authentication step, eliminating the chance of misusing an exposed API key. "Read more":/core-features/authentication#selecting-auth about selecting an authentication mechanism.
+
+h3(#ensuring-tokens-are-auto-renewable). Ensuring tokens are auto-renewable
+
+Tokens are short-lived i.e. they expire after a given time. It is almost never correct to provide a token to a client library without ensuring the client library has a means to get a new token automatically when it needs one. The Ably client library SDKs provide two mechanisms in the "ClientOptions":/realtime/authentication#token-authentication so that they can obtain tokens automatically using either "@authUrl@":/realtime/authentication#token-upgrading or the "@authCallback@":/realtime/authentication#token-upgrading.
 
 h3(#rest-or-realtime). Realtime vs REST
 
-Ably provides two types of client libraries. "Realtime":/realtime libraries maintain a persistent connection to Ably and can be considered stateful, whereas "REST":/rest libraries issue HTTP requests to the Ably platform and are stateless.
+Ably Client Library SDKs may implement two interfaces: realtime and REST. The "Realtime":/realtime  interface maintains a persistent connection to Ably and can be considered stateful, whereas the "REST":/rest interface issues HTTP requests to the Ably platform and is stateless. The REST interface is therefore much lighter weight as it does not need the complexity of maintaining state. The REST interface can also only publish, but not subscribe to messages.
```

Comment:
```suggestion
Ably Client Library SDKs may implement two interfaces: realtime and REST. The "Realtime":/realtime interface maintains a persistent connection to Ably and can be considered stateful. The "REST":/rest interface issues HTTP requests to the Ably platform and is stateless. The REST interface is therefore lighter weight because it does not need to maintain state. 
```

(I suggest removing the REST/pub only limitation here, because you explain it very succinctly later on in this section)